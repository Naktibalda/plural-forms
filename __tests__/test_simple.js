import { getFormula, getNPlurals, getPluralFunc, hasLang,
    getAvailLangs, getPluralFormsHeader } from '../dist/simple';

test('test getFormula for en locale', () => {
    expect(getFormula('en')).toBe('(n !== 1)');
});

test('test getNPlurals for en locale', () => {
    expect(getNPlurals('en')).toBe(2);
});

test('test getPluralFunc for en locale', () => {
    const fn = getPluralFunc('en');
    expect(fn(0, ['banana', 'bananas'])).toBe('bananas');
    expect(fn(1, ['banana', 'bananas'])).toBe('banana');
});

test('test hasLang', () => {
    expect(hasLang('en')).toBe(true);
    expect(hasLang('zzz')).toBe(false);
});

test('test getAvailLangs', () => {
    expect(getAvailLangs(true)).toMatchSnapshot();
});

test('test getPluralFormsHeader', () => {
    expect(getPluralFormsHeader('en')).toBe('nplurals = 2; plural = (n !== 1)');
});
