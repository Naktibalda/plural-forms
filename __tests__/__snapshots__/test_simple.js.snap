// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test printAvailableLocales 1`] = `
"Acholi - ach
Afrikaans - af
Akan - ak
Amharic - am
Aragonese - an
Arabic - ar
Mapudungun - arn
Asturian - ast
Aymar√° - ay
Azerbaijani - az
Belarusian - be
Bulgarian - bg
Bengali - bn
Tibetan - bo
Breton - br
Bodo - brx
Bosnian - bs
Catalan - ca
Chiga - cgg
Czech - cs
Kashubian - csb
Welsh - cy
Danish - da
German - de
Dogri - doi
Dzongkha - dz
Greek - el
English - en
Esperanto - eo
Spanish - es
Estonian - et
Basque - eu
Persian - fa
Fulah - ff
Finnish - fi
Filipino - fil
Faroese - fo
French - fr
Friulian - fur
Frisian - fy
Irish - ga
Scottish Gaelic - gd
Galician - gl
Gujarati - gu
Gun - gun
Hausa - ha
Hebrew - he
Hindi - hi
Chhattisgarhi - hne
Croatian - hr
Hungarian - hu
Armenian - hy
Indonesian - id
Icelandic - is
Italian - it
Japanese - ja
Lojban - jbo
Javanese - jv
Georgian - ka
Kazakh - kk
Khmer - km
Kannada - kn
Korean - ko
Kurdish - ku
Cornish - kw
Kyrgyz - ky
Letzeburgesch - lb
Lingala - ln
Lao - lo
Lithuanian - lt
Latvian - lv
Maithili - mai
Mauritian Creole - mfe
Malagasy - mg
Maori - mi
Macedonian - mk
Malayalam - ml
Mongolian - mn
Manipuri - mni
Mandinka - mnk
Marathi - mr
Malay - ms
Maltese - mt
Burmese - my
Nahuatl - nah
Neapolitan - nap
Norwegian Bokmal - nb
Nepali - ne
Dutch - nl
Norwegian Nynorsk - nn
Norwegian - no
Northern Sotho - nso
Occitan - oc
Oriya - or
Punjabi - pa
Papiamento - pap
Polish - pl
Piemontese - pms
Pashto - ps
Portuguese - pt
Romansh - rm
Romanian - ro
Russian - ru
Kinyarwanda - rw
Yakut - sah
Santali - sat
Scots - sco
Sindhi - sd
Northern Sami - se
Sinhala - si
Slovak - sk
Slovenian - sl
Somali - so
Songhay - son
Albanian - sq
Serbian - sr
Sundanese - su
Swedish - sv
Swahili - sw
Tamil - ta
Telugu - te
Tajik - tg
Thai - th
Tigrinya - ti
Turkmen - tk
Turkish - tr
Tatar - tt
Uyghur - ug
Ukrainian - uk
Urdu - ur
Uzbek - uz
Vietnamese - vi
Walloon - wa
Wolof - wo
Yoruba - yo
Chinese - zh"
`;
